### PR TITLE
feat: add prisma SQL migration for job-run internal flag backfill

### LINKED ISSUE #557: [TRI-1366] fix: JobRun needs an internal boolean column, to reflect the value in Job
We use internal jobs for things like webhook registration, and those jobs have `internal = true` set in the database. We also need to add an `internal` column to JobRun that is set to `true` when the job it's associated with is `internal = true`. This is just denormalization, so when we do usage queries (in the new usage page) we can easily eliminate internal runs from the usage. The migration for this should upgrade existing runs to the correct `internal` value.

<sub>[TRI-1366](https://linear.app/triggerdotdev/issue/TRI-1366/fix-jobrun-needs-an-internal-boolean-column-to-reflect-the-value-in)</sub>

### PR DESCRIPTION
**Pull request by hmacr**
Closes #557

This builds on top of #563 and should ideally be merged after #563 goes live.

The intention is to separate the schema and API change from the backfill script. This way, the jobs & job runs that get created during the backfill execution can still have the correct `internal` flags set (the API takes care of it).

## ✅ Checklist

- [x] I have followed every step in the [contributing guide](https://github.com/triggerdotdev/trigger.dev/blob/main/CONTRIBUTING.md)
- [x] The PR title follows the convention.
- [x] I ran and tested the code works

---

## Testing

- Verified the past runs of internal jobs have their job-run internal flag set to `true`.
- Verified that the migration runs only once.

---

## Changelog

- Add a backfill script that sets the `internal` flag for past `JobRun` entries.
- To avoid DB overload, chunks of jobs (100 - configurable) are used to load their associated job-runs. 

💯

### CODE REVIEW

**Comment by hmacr** [comment]
Marking it as draft, since it has dependency on #563.
Needs to be rebased with the main branch once #563 is merged.

File: apps/webapp/prisma/migrations/jobRunInternal.ts [hunk 1]
```diff
@@ -0,0 +1,74 @@
+import { prisma, PrismaClientOrTransaction } from "~/db.server";
+import chunk from "lodash.chunk";
+
+const MIGRATION_NAME = "2023-10-05-SetJobRunInternalFlag";
+const JOB_CHUNK_SIZE = 100;
```
Comment (hmacr) [inline apps/webapp/prisma/migrations/jobRunInternal.ts]: What should be the ideal chunk size?

**Comment by ericallam** [comment]
I like the idea of doing the backfilling separately, but I think we should try do the backfill in SQL using Prisma migrations. Something like: 

```sql
UPDATE "JobRun"
SET "internal" = "Job"."internal"
FROM "Job"
WHERE "JobRun"."jobId" = "Job"."id";
```

**Comment by hmacr** [comment]
Ah ok. I was worried that a large update would reduce the DB throughput for other processes, so wrote a backfill script.

I'll update the PR with the necessary SQL.

**Comment by hmacr** [comment]
I've removed the backfill script and added the SQL to update the job-run internal flag.

The migration SQL worked as expected on local.

**Comment by hmacr** [comment]
Closing since this code is moved to #563

**Pull request closed without merging**